projects/core/src/configurator/commons/facade/configurator-groups.service.ts

```diff
@@ -22,20 +22,82 @@ export class ConfiguratorGroupsService {
     private configuratorGroupStatusService: ConfiguratorGroupStatusService
   ) {}
 
+  /**
+   * Returns the current group Id.
+   * In case no group Id is being set before returns the first group of the configuration.
+   * Return null when configuration contains no groups.
+   *
+   * @param owner configuration owner
+   */
   getCurrentGroupId(owner: GenericConfigurator.Owner): Observable<string> {
     return this.configuratorCommonsService.getConfiguration(owner).pipe(
       map((configuration) => {
         if (configuration?.interactionState?.currentGroup) {
           return configuration.interactionState.currentGroup;
         } else {
-          return configuration?.groups?.length > 0
-            ? configuration.groups[0].id
+          return configuration?.flatGroups?.length > 0
+            ? configuration.flatGroups[0].id
             : null;
         }
       })
     );
   }
 
+  /**
+   * Return the first conflict group of a configuration.
+   *
+   * @param configuration - Configuration
+   */
+  getFirstConflictGroup(
+    configuration: Configurator.Configuration
+  ): Configurator.Group {
+    return configuration.flatGroups.find(
+      (group) => group.groupType === Configurator.GroupType.CONFLICT_GROUP
+    );
+  }
+
+  /**
+   * Navigates to the first non-conflict group of the configuration which is not completed.
+   * This method assumes that the configuration has uncompleted groups,
+   * the caller has to verify this prior to calling this method.
+   *
+   * @param owner - Configuration owner
+   */
+  goToFirstUncompletedGroup(owner: GenericConfigurator.Owner): void {
+    this.configuratorCommonsService
+      .getConfiguration(owner)
+      .pipe(take(1))
+      .subscribe((configuration) =>
+        this.navigateToGroup(
+          configuration,
+          this.configuratorGroupStatusService.getFirstUncompletedGroup(
+            configuration
+          )?.id,
+          true
+        )
+      );
+  }
+
+  /**
+   * Navigates to the first conflict group and sets the conflict header as parent group.
+   * This method assumes that the configuration has conflicts,
+   * the caller has to verify this prior to calling this method.
+   *
+   * @param owner Configuration Owner
+   */
+  goToConflictSolver(owner: GenericConfigurator.Owner): void {
+    this.configuratorCommonsService
+      .getConfiguration(owner)
+      .pipe(take(1))
+      .subscribe((configuration) =>
+        this.navigateToGroup(
```

Comment:
You mix up **goTo** and **navigateTo**. Semantically both are correct but please decide yourself for either **goTo** or **navigateTo**. I would prefer **navigateTo** but it is up to you what you choose

Reply:
Done.